src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/StraxCoinviewRule.cs

```diff
@@ -106,5 +125,32 @@ protected override void AllowSpend(TxOut prevOut, Transaction tx)
 
             // Otherwise allow the spend (do nothing).
         }
+
+        /// <inheritdoc />
+        public override async Task RunAsync(RuleContext context)
+        {
+            if (this.logic != null && context.ValidationContext.BlockToValidate.Header is PosBlockHeader posHeader && posHeader.HasSmartContractFields)
+                await this.logic.RunAsync(base.RunAsync, context);
```

Comment:
please add ConfigureAwait(false) everywhere where await is

Reply:
done